File: src/main/java/io/github/jho951/assignment/job/web/dto/ImageJobSummaryResponse.java
package io.github.jho951.assignment.job.web.dto;

import java.time.Instant;

public record ImageJobSummaryResponse(
        String jobId,
        String status,
        String imageUrl,
        int attemptCount,
        Instant createdAt,
        Instant updatedAt,
        Instant completedAt,
        Instant expiresAt,
        ImageJobErrorResponse error
) {
}

File: src/main/kotlin/io/github/jho951/assignment/job/web/dto/ImageJobSummaryResponse.kt
package io.github.jho951.assignment.job.web.dto

import java.time.Instant
import kotlin.jvm.JvmRecord

@JvmRecord
data class ImageJobSummaryResponse(
    val jobId: String,
    val status: String,
    val imageUrl: String,
    val attemptCount: Int,
    val createdAt: Instant?,
    val updatedAt: Instant?,
    val completedAt: Instant?,
    val expiresAt: Instant?,
    val error: ImageJobErrorResponse?
)
